src/generic/UnivPoly.jl

```diff
@@ -1004,6 +1004,21 @@ end
 #
 ###############################################################################
 
+function zero!(a::UnivPoly{T}) where {T <: RingElement}
+   a.p = zero!(a.p)
+   return a
+end
+
+function one!(a::UnivPoly{T}) where {T <: RingElement}
+   a.p = one!(a.p)
+   return a
+end
+
+function neg!(z::UnivPoly{T}, a::UnivPoly{T}) where {T <: RingElement}
+   z.p = neg!(a.p)
```

Comment:
this needs to be 
```julia
z.p = neg!(z.p, a.p)
```
to not modify `a`. I'll add this patch to https://github.com/Nemocas/AbstractAlgebra.jl/pull/1847